### ISSUE TITLE
Support for unstructured grid with a vertical dimension

### ISSUE THREAD

**Issue by dennissergeev** [bjlittle/geovista#15]
## ✨ Feature Request

Firstly, thanks very much for creating this library, I'm excited to use it in my research!

Secondly, would it be possible to generalise `geovista.Transform.from_unstructured()` for it to take in arrays with more than 2 dimensions, i.e. vertical levels (and possibly time)?

## Motivation

I am trying to visualise LFRic output (happy to eventually contribute to the gallery by the way), and while the `from_unstructured()` method works great for 2D arrays, I would like to plot something w.r.t. model height. Currently there's no obvious way to do this as that function accepts only longitudes and latitudes, throwing an error if I pass a full 3D array as `data`.

This is probably related to the ["isosurfaces support" to-do item](https://github.com/bjlittle/geovista/projects/1#card-61044978).

**Comment by bjlittle** [comment]
Hey @dennissergeev,

Stoked to hear that you're excited about `geovista`... I am too!

Prior to `geovista`, I was playing with animating model time-series and also providing a slider so the user can manually control the render of spatial data with another single dimension (such as time). To be honest, that worked reasonably well. Here's a proof-of-concept I'm currently working on for `geovista`... it's show-casing an unstructured ocean model with variable resolution quad-faces:

![scmlite](https://user-images.githubusercontent.com/2051656/144131248-c0c9c7e9-4e9a-4aaa-a677-422528903321.gif)

Apologies for the poor quality GIF, it doesn't really do the colormap justice... but I think you get the idea of the direction I'm headed in.

So time-series support is definitely in the pipeline, perhaps along with `ipywidget` integration for jupyter... but the overriding goal here is for `geovista` to easily integrate into the existing scientific python ecosystem e.g., scripting, notebooks, https://panel.holoviz.org/ et al... and this should be a level playing field for the community, with regards to whether you're coming to `geovista` from `iris`, `xarray`, `cupy` or just where ever.

There are some fundamentals that I'm keen to bank for `geovista` such as CRS, planar projection and texture mapping support, but later isosurfaces, isolines, streamplots, winds visualization, mesh warping for ocean bathymetry and surface altitudes, web services tiling etc are all in play :+1: 

With regards to the vertical I've not quite thought that far ahead to be honest (`geovista` is just born, so I'm focusing on a solid core first and building outwards). Nevertheless, what kinda viz are you particularly interested in?

I was mulling over geodesic ribbon trajectory slices through unstructured meshes to show vertical levels in the planar, but I've not ventured too far into this space... but I am motivated by concrete use cases though :wink: 

So if you can be more specific and give me some examples this would really help :+1:

**Comment by dennissergeev** [comment]
Thanks for the speedy response @bjlittle!

> Here's a proof-of-concept I'm currently working on for geovista... 

Looks awesome! I would love to use this once you're happy with the prototype.

> the overriding goal here is for geovista to easily integrate into the existing scientific python ecosystem

I agree that's the way to go. I'm mostly using `iris` because I mostly use the UM in my research (and now starting to use LFRic).

> There are some fundamentals that I'm keen to bank first for geovista

Fair enough. I can try to help with some "good first-time issues"...

> Nevertheless, what kinda viz are you particularly interested in?

I can of course just plot some slices of the vertical coordinate, but eventually I would like to create plots with the vertical dimension, i.e. planet's radius, such as vertical cross-sections, isosurfaces, wind vectors. As an example, here's what I made with PyVista a while ago: https://dennissergeev.github.io/exoconvection-apj-2020/

**Comment by bjlittle** [comment]
@dennissergeev Awesome, I've come across that visualization before, super cool, congrats!

Okay, I can see what you're aiming for, and that all makes sense, thanks. I'm guessing your Trappist-1e 3D viz could also be projected e.g., say the sphere to Plate Carree, but still have the 3D vertical convection layers on top, thus taking advantage of a 2D planar but viewed in 3D, if you get my meaning :thinking: 

I think that you're original 3D viz is perhaps currently possible in ParaView. I know that @tinyendian (Wolfgang Hayek @ NIWA) might be able to comment or point you to what's possible there too - if you're interested?

**Comment by dennissergeev** [comment]
Thanks @bjlittle!

> I'm guessing your Trappist-1e 3D viz could also be projected e.g., say the sphere to Plate Carree,
Yes, certainly. Visualising it both in a planar projection or as a 3D sphere would be great! I guess that's why you're envisaging CRS support - to have an API similar to `cartopy`?

> I think that you're original 3D viz is perhaps currently possible in ParaView.
That would be useful to know, so any tips from Wolfgang would be appreciated. 
At the same time, this is not urgent and I'm happy to wait/contribute for geovista to mature enough to support this.

P.S. Great logo by the way!

**Comment by tinyendian** [comment]
Hi @dennissergeev and @bjlittle, visualisations such as Trappist-1e (which looks great btw!!) can definitely be done with ParaView, including time-dependence of the model data, spherical projection, moving cameras, interactive web publication, ...

I haven't used PyVista before, but, looking at the examples, it seems to use a similar philosophy as ParaView, providing a simplifying layer on top VTK. ParaView comes with a few perks, such as a powerful GUI, and a client-server mode, which is useful for very large models that require an HPC backend to handle. Its Python scripting capabilities are comprehensive and integrate nicely with the GUI, but the price to pay is that the scripts look somewhat obscure compared to PyVista, the latter seems a bit more Pythonic.

Very happy to help, if you'd like to give ParaView a try (it can read CF-netCDF and LFRic output), or if you have any questions.

Btw, I worked in the exoplanet group in Exeter 10 years ago as a postdoc, great to see the work that you guys are doing there! And say hi to Nathan for me please 🙂.

**Comment by dennissergeev** [comment]
hi @tinyendian,

Thanks! Yes, the exoplanet group is growing stronger, so expect more cool 3D visualisations of exoplanet simulations :smile:!

I actually used ParaView a little bit in the past, though only as a GUI. Now I prefer using PyVista because of its pythonic API and integration with Jupyter, but in the meantime if I wanted to give ParaView another go and visualise LFRic output, where should I start?

**Comment by tinyendian** [comment]
Hi @dennissergeev, the easiest way to start with ParaView is a conda installation of the LFRic reader plugin, which will install the latest ParaView GUI as a dependency,
```
conda install -c conda-forge lfric_reader
```
If you have a ParaView build on your system already (along with the required build tools and dependencies), you can also easily build the reader yourself, see the instructions on https://github.com/niwa/lfric_reader.

Note that Jupyter integration was recently added to ParaView, too, https://blog.kitware.com/paraview-jupyter-notebook/, but I haven't tried this out yet.

With regards to Python scripting, I agree that PyVista coding looks a lot nicer and straightforward. ParaView really has its strengths when it comes to interactive visualisation (have a look at, e.g., linked cameras for comparing two models), and handling very large datasets (hundreds of millions of cells) efficiently. Maybe there will be ways to integrate Iris/PyVista and ParaView a bit in the future, e.g., via automatically generated ParaView scripts (ParaView can be fully controlled by Python scripting).

Looking forward to seeing more exoplanet simulations 😍!

**Comment by dennissergeev** [comment]
Thanks very much @tinyendian! I'll give it a go.

**Comment by bjlittle** [comment]
Hey @dennissergeev,

Just wanted to let you know that I want to get my ducks-in-a-row to schedule time for supporting this issue. The first step in this journey will be to extend the capability of the `geovista.bridge` to ingest mesh data with a *vertical* dimension.

I'm guessing that you've got a tonne of archived data that you'd be keen for `geovista` to visualise out-the-box... is it possible for you to share it with me somehow?

Cheers :smile:

**Comment by dennissergeev** [comment]
Hey @bjlittle!

This is exciting to hear! Happy to help test / develop things, just ping me.
And I'm happy to supply some sample data, of course. Here's an output file from one of my recent simulations:
https://we.tl/t-gt1W6B4mkl

**Comment by bjlittle** [comment]
@dennissergeev Starting to play in this space, so just I thought that I'd give you a minor update...

I've been working on a use case with an oceanographer at the Met Office, where they have 50 depth levels (bathymetry) of ORCA ocean model data (on tri-polar curvilinear grid), which has been pre-processed through a [canny edge filter](https://en.wikipedia.org/wiki/Canny_edge_detector) to detect potential temperature gradients. Initially, they want to visualise the data as a point cloud, which is an ideal first use case.

I was able to easily extend `geovista` to create the geo-referenced point cloud, which kinda looks like this:

- I've rendered the points with their levels values
- Used the [cmocean deep](https://matplotlib.org/cmocean/#deep) colormap, which is super lush
- Added a texture map base layer with opacity (to confirm geo-refrencing)
- Added 10m Natural Earth coastlines (to confirm geo-refrencing)

https://user-images.githubusercontent.com/2051656/221884113-62c07148-d4c1-4fb9-a4a1-0917cd013ef1.mp4

Next steps are to:

- make it easy to reproject the point cloud (say to Plate Carree), which might make interactive interrogation a wee bit more intuitive
- extend to support non-`polydata`, and integrate into the `geovista.bridge` API

In hindsight, I should have rendered this example with `culling=back` to make the things slightly less confusing, but this is just a PoC at the moment. 

Baby steps.

**Comment by bjlittle** [comment]
https://user-images.githubusercontent.com/2051656/221891145-52d14e16-8f88-4c7b-85d7-110b2f9864f7.mp4

**Comment by bjlittle** [comment]
https://user-images.githubusercontent.com/2051656/221891409-bfc2ddd4-56ef-4085-a0be-04d4475393dd.mp4

**Comment by dennissergeev** [comment]
Sorry for my slow response, I was on holiday :)

This looks amazing, many thanks for sharing! Is this functionality available in the latest release? (No rush, just wondering.)

And since you've shown that point cloud rendering is possible, I guess extending this to vectors/arrows to show the flow velocity in 3D should be quite straightforward, right?

**Comment by bjlittle** [comment]
Hey @dennissergeev,

Hope you had an awesome holiday.

This functionality will be available in the forthcoming 0.2.0 release.

Rendering point clouds is a nice entry point into this space, so 3D grids and vectors should follow, yeah.

There are a few other things that I also need to tidy-up, so I'm going to release little and often, if I can. But I'll certainly ping you when I make more progress :+1:

**Comment by dennissergeev** [comment]
Excellent, thank you @bjlittle!

**Comment by dennissergeev** [comment]
Hi @bjlittle! Great to see the steady pace of `geovista` releases :smile: 
Does the latest release include this functionality or did you move it to the next release?
Cheers!

**Comment by bjlittle** [comment]
Hey @dennissergeev,

Lovely to hear from you again, as always. Great question :smile: 

Okay, so the `0.2.0` point-cloud support is the first step in that direction, but ultimately it's not what you actually want.

My focus at the moment is banking stable core functionality with the view to presenting a lightning talk at [SciPy 2023](https://www.scipy2023.scipy.org/) in July, where at that point `geovista` finally steps out from the shadows. I'm really interested to see the initial reaction from the scientific community to `geovista`. I think it's going to be a wee bit of a watershed moment, which I'm equally nervous and excited about... 

Anyways, in terms of what's roughly coming up next and when:

- the main deliverable for `0.3.0` will contain support for projecting lines and coastlines, including the ability to add a configurable graticule (i.e., labelled parallels and meridians) to the mesh to aid visual geolocation
- `0.4.0` will be all about bootstrapping the docs. For me, the docs are the shop window to `geovista`, so having some decent, quality docs is paramount prior to SciPy.

Both `0.3.0` and `0.4.0` are going to follow on the coattails of `0.2.0` and land prior to SciPy.

If there's still time before SciPy, I'd love to steamline and expose a slicker projection API and also seed patterns for `geovista.crs` projection classes e.g., address #140 

Realistically, post SciPy when the dust settles I'm super keen to finally address this issue.

In my head, that timeline and schedule makes sense... is that okay with you?

**Comment by dennissergeev** [comment]
Yep, totally okay with me! Good luck at SciPy, I'm confident people will be super excited about `geovista` - I know my colleagues are!

And do let me know if you want me to test/review anything.

**Comment by jbusecke** [comment]
Hey everyone, I just came here to
a) thank you for this amazing package. I just made some [videos](https://vimeo.com/manage/videos/863996198) for a project site visit and they look stunning!
b) add a gigantic +1 to supporting 3D (lon, lat, depth) visuzlization. My usual usecases revolve around isosurfaces. This [video](https://vimeo.com/manage/videos/783450053) was done with pyvista-xarray and my first time for me that 3D visualization of my usual ocean/climate model data 'just worked'. Having this sort of plot (with bathymetry) on a 3D globe would be monumental for my work. 
Further out, having a way to call this from pyvista-xarray somehow would be the holy grail for the researchers/students in my project but I understand that the basics need to work first 🤗

If there is anything I could do to help you out here, please let me know. Unfortunately I am not well versed in pyvista (and the underlying stack) yet, but could certainly test drive, provide data samples etc. 

Again big thanks for this awesome package.

**Comment by dennissergeev** [comment]
Amazing videos @jbusecke!

**Comment by bjlittle** [comment]
Thanks so much @jbusecke. Your feedback means a lot and your animations are truly lush 🤩

`geovista` is really just starting on it's journey, and I'm so glad you found us. Sounds like your use case definitely bolsters the demand for volume rendering for sure 👍 

After hanging out with @banesullivan at SciPy 2023, we're keen to join the dots between `pyvista-xarray` and `geovista` to open the door for the xarray community to easily render and explore their Earth sciences data using `pyvista`. I agree with you; the potential is huge.

I also had the pleasure of crossing paths with @mgrover1, who similarly wants to render volumes for radar data... so it's inevitable that this is going to happen, which I'm pretty darn excited about too.

Game on.

**Comment by mgrover1** [comment]
I would be interested in a sprint on this soon 😄

**Comment by tkoyama010** [comment]
@jbusecke
It's a great video. Could you post this on the PyVista forum to introduce it to more people? If you can't take the time to do so I will introduce that you created this video and post it.
https://github.com/pyvista/pyvista/discussions/new?category=show-and-tell

**Comment by bjlittle** [comment]
@dennissergeev @jbusecke @mgrover1 

Quick question ... when we get to the point where we have a proof-of-concept for the volume rendering in a feature branch (I'm going to figure out when I can schedule this work), would you guys be happy to kick the tyres to test/play and feedback?

Thumbs-up if you're in 👍

**Comment by jbusecke** [comment]
> @jbusecke
> 
> It's a great video. Could you post this on the PyVista forum to introduce it to more people? If you can't take the time to do so I will introduce that you created this video and post it.
> 
> https://github.com/pyvista/pyvista/discussions/new?category=show-and-tell

Sure thing will do that next week.

**Comment by jbusecke** [comment]
> Prior to that it would be ideal to get my hands on some representative data that you wanted to render. Nothing high resolution, if possible, just some data that I can quickly render when developing/testing. Cheers 🍻

@bjlittle I am thinking mostly about CMIP6 climate model output (this should be very representative for most climate/ocean model output. If you can run this snippet

```python
from xmip.utils import google_cmip_col
from xmip.preprocessing import combined_preprocessing
col = google_cmip_col()
cat = col.search(
    variable_id='so',
    table_id='Omon',
    source_id='CanESM5',
    experiment_id='historical',
    member_id='r1i1p1f1'
)
ds = cat.to_dataset_dict(preprocess=combined_preprocessing)['CMIP.CCCma.CanESM5.historical.Omon.gn']
ds = ds.isel(time=0).squeeze().drop([co for co in ds.coords if co not in ['lon', 'lat']])
ds.to_netcdf('geovista_cmip_demo.nc')
```
You could get a ~20 MB sample of 3D ocean salinity. Let me know if this works (requires [xMIP](https://github.com/jbusecke/xMIP)) otherwise Ill see how to send you this data.

@tkoyama010 https://github.com/pyvista/pyvista/discussions/4937

**Comment by tkoyama010** [comment]
This issue is still important.

**Comment by bjlittle** [comment]
@all-contributors please add @dennissergeev for ideas

**Comment by bjlittle** [comment]
@all-contributors please add @mgrover1 for ideas and promotion

**Comment by edmundhenley-mo** [comment]
> Quick question ... when we get to the point where we have a proof-of-concept for the volume rendering in a feature branch (I'm going to figure out when I can schedule this work), would you guys be happy to kick the tyres to test/play and feedback?

@bjlittle - just found this one - subscribed immediately!
TLDR: once you've got your proof-of-concept done, shout iff you want to extend onwards-and-upwards for some space weather use-cases! Happy to help with some data wrangling & tyre kicking if so - no worries if you've got hands full with use-cases from lower down!

<details>

<summary>Click for context, and potential good 1st target = WACCM-X</summary>

## Context
This sort of volume / isosurface rendering could be very interesting for space weather use-cases in Earth's ionosphere, thermosphere and below (where we have more-easily supportable geographic lat/lon coords, as well as more bonkers ones too!). These regions can be highly dynamic, with multi-scale spatiotemporal effects which ~require seeing ~4D representations to understand full picture - risk losing insights if you collapse data straight to 2D or lower. 
Geovista's interactivity - ability to move round from dayside to nightside, or look down on the poles - could be super-useful for enabling workflows which start with exploration of the full-fat datasets.
* *i.e. basically exactly the same points @dennissergeev and Cristina made in lightning talks on Geovista!*

There's some very nice videos here illustrating this sort of thing - look at all of those ripples as the solar storm hits on 17 Mar 2015 - the 4D synoptic view here invaluable - a lower-dimensional representation couldn't cut it!
* [Thermospheric temperature](https://www.youtube.com/watch?v=QmXE-UemKw0)
* [Ionospheric electron number density](https://www.youtube.com/watch?v=Df4wCKyNpN4) - volumetric equivalent of the [total electron content in iris examples](https://scitools-iris.readthedocs.io/en/latest/generated/gallery/meteorology/plot_TEC.html#ionosphere-space-weather)
* More details on what's being plotted in [Figure 1](https://angeo.copernicus.org/articles/39/189/2021/#section7&gid=1&pid=1) of [Palmroth+ 2021](https://doi.org/10.5194/angeo-39-189-2021)

From dim recollection I believe @eelcodoornbos generated these with Blender (?). In any case, using output from NCAR's [WACCM-X model](https://www2.hao.ucar.edu/modeling/waccm-x).

## WACCM-X: a good first target?
Iff this is of interest, WACCM-X might be a good target for dipping geovista's toes in the space weather water (if you've not already?):
* it's a state-of-the-art community model, coupling into lower atmosphere ([Liu+ 2017](http://dx.doi.org/10.1002/2017MS001232))
* well-behaved: nice netCDF output, on sane geographic grid, vertical on pressure levels - and data volumes not too big by terrestrial standards, so good for exploration (characteristics in [slide 5 of WACCM-X tutorial](https://www2.hao.ucar.edu/sites/default/files/2021-12/WACCM-Xtutorial.pdf)
* so good place to start!
* *Note NCAR's [TIE-GCM](https://www.hao.ucar.edu/modeling/tgcm/tie.php) another good contender: similar characteristics, and very widely used*

Later can see if Geovista's ugrid support might be an easy way around trickier things like geomagnetic coordinates in other models, ...

There's WACCM-X output available from the NASA Community Coordinated Modeling Center (CCMC). E.g. for this "St Patrick's day storm", there's run ["WACCMX-Weimer-01_2015-03-TP-01_102523_IT_1"](https://ccmc.gsfc.nasa.gov/results/viewrun.php?runnumber=WACCMX-Weimer-01_2015-03-TP-01_102523_IT_1), and from there:
* [Data download](https://ccmc.gsfc.nasa.gov/results/run_files.php?runnumber=WACCMX-Heelis-01_2015-03-TP-01_021824_IT_1)
* [Vis API](https://ccmc.gsfc.nasa.gov/cgi-bin/CCMCVis_3D.cgi?dir=33759)

Happy to put some effort into giving this a whirl if of interest!

</details>

**Comment by bjlittle** [comment]
@all-contributors please add @edmundhenley-mo for ideas and userTesting

**Comment by bjlittle** [comment]
@edmundhenley-mo Great! :rocket: 

Good news! I've finally managed to secure a deployment into a science team to work on this ... it's the only way I'm going to secure the time to do it!

I'll come knocking at your door afterwards for sure! :+1:

**Comment by dennissergeev** [comment]
Amazing 🤩

**Comment by mgrover1** [comment]
This is INCREDIBLE! Great work here @bjlittle 👍

**Comment by bjlittle** [comment]
Same dataset but with isosurfaces and an opacity linear transfer function for plume-like rendering ...

![fukushima-contour-opacity](https://github.com/user-attachments/assets/642d5fae-bf73-43bf-8cea-a69cf513020d)

The widgets won't be part of `geovista`, they just make it convenient to quickly alter what and how geometries are being rendered :+1:

**Comment by dennissergeev** [comment]
Wow this is amazing!!! Isosurfaces are the main thing I'm waiting for in geovista, so thank you so much for working on this!

**Comment by mo-benjaminevans** [comment]
This looks great @bjlittle. It will help with expert judgment on plume dispersion, engagament and model validation etc. In regards this issue and #16 it would be great to be able to overlay 3D vectors on top of the 3D plume. Is the plan that both issues/PRs should work together?

**Comment by bjlittle** [comment]
@mo-benjaminevans Yes, indeed.

Both #15 and #16 completely complement each other. Being able to visualize how winds influence plume behaviour is a critical part of your science. So just to confirm, you will be able to overlay 3D vectors on top of a 3D plume :+1: 

You'll be glad to hear that the plan is for both features to be available in the next release of `geovista` 0.6.0.

**Comment by pp-mo** [comment]
This is quite exciting.
I purposely omitted 3d from my "UGRID conformance rules" [proposal](https://github.com/ugrid-conventions/ugrid-conventions/pull/54)   (now included in published conventions [here](https://ugrid-conventions.github.io/ugrid-conventions/conformance/)) because I couldn't see how to make the existing statements hang together.
It has even been proposed that we might [pause the whole 3D thing for UGRID since it had **so far seen little interest**](https://github.com/ugrid-conventions/ugrid-conventions/pull/54#issuecomment-1343275483),  
But perhaps practical efforts like this can show the way to making it a real thing.

**Comment by jmb64** [comment]
Hi,
    I must start by saluting the work of @bjlittle, @banesullivan, and all the developers
    and contributors to the pyvista/geovista endeavour !
    I've been using this software for a little while now and read this issue 15 (and others)
    with great interest. Posts by @jbusecke, @denissergeev really resonate with my own experience.
    I have been involved in many different projects in my career (land surface processes, atmospheric
    data assimilation based on adjoint methods, development of our global coupled atmosphere-ocean
    operational NWP forecast model here at ECCC, ... All along the way I always considered the need for
    validation, diagnostics and visualization tools to pivotal role in the success of the scientific
    journey. I was amazed a couple of years ago with the advent of panel/holoviz and the possibility
    to build small interactive visualization application instead of relying on the generation of zillions
    of static images. Now I feel that 3D isosurfaces, real-time 3D direct volume rendering are  
    tools that are within reach of scientific like me with no solid expertise in the field. Using
    pyvista gave me the opportunity to discover VTK (even though it's been there for over 30 years !)
    and ParaView. I wish this happened earlier in my career but ... From my neophyte's perspective it
    seems that there is a convergence of factors happening (ex.: hardware:NVIDIA's RTX cores,
    software:ANARI standard API for 3D rendering, rise of AI and gaming engines, a multiplication of
    rendering backends and a consciousness that this technology has been hidden most of the time
    in software frameworks of (academic) experts (with not bad intention but a lack of time and resources).
    A last thing I would like to add, in searching for for volume rendering application one particular
    feature (or lack of) is the georeferencing information (coastlines, lat, lon, height, depth, pressure,...).
    I have documented a few initiatives over the years (ParaView) but none seem to have stuck. In geophysics
    ( I deal with ocean and atmospheric datasets) this missing piece is really annoying. I was really please
    to discover that geovista had this in mind (lat, lon graticules).
    Well, I guess this is how I wanted to add my support to this work with both hands !
    
    Oh, I almost forgot. I was wondering if @jbusecke the python code for the Pacific OMZ isosurfaces
    (https://vimeo.com/783450053) was public somewhere ? :-)

    Cheers,
    Jean-Marc

**Comment by banesullivan** [comment]
> Oh, I almost forgot. I was wondering if @jbusecke the python code for the Pacific OMZ isosurfaces
(https://vimeo.com/783450053) was public somewhere ? :-)

Reposting this last bit so that it pings Julius: is this code open by chance? 😄

**Comment by jmb64** [comment]
Thanks @banesullivan. 
I figured the first ping would do it but then ... maybe not !  :-)

**Comment by banesullivan** [comment]
Oh it totally did 🤦🏻‍♂️ I didn't see the first `@` tag at the top of the message. I'm just extra eager to see if the code is available 😄

**Comment by jbusecke** [comment]
I refactored the code and added it to this [gallery repo](https://github.com/jbusecke/pv-xarray-gallery?tab=readme-ov-file#data-examples).

**Comment by jmb64** [comment]
@jbusecke, I'm just returning from summer vacation. Good to know you are back in shape. 
I will certainly have a look at your examples.
Thanks!

**Comment by dennissergeev** [comment]
still important...
